client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java

```diff
@@ -1043,6 +1043,11 @@ public FindBrokerResult findBrokerAddressInSubscribe(
             slave = brokerId != MixAll.MASTER_ID;
             found = brokerAddr != null;
 
+            if (!found && slave) {
+                brokerAddr = map.get(brokerId + 1);
```

Comment:
How do we handle if the slave has happened to become a master?

Reply:
The broker id will change as dledger role change